nuxeo-features/nuxeo-automation/nuxeo-automation-test/src/test/java/org/nuxeo/ecm/automation/server/test/AsyncOperationAdapterTest.java

```diff
@@ -0,0 +1,202 @@
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Nelson Silva <nsilva@nuxeo.com>
+ */
+package org.nuxeo.ecm.automation.server.test;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.io.FileUtils;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.automation.client.Constants;
+import org.nuxeo.ecm.automation.client.RemoteException;
+import org.nuxeo.ecm.automation.client.Session;
+import org.nuxeo.ecm.automation.client.adapters.AsyncSession;
+import org.nuxeo.ecm.automation.client.model.Blob;
+import org.nuxeo.ecm.automation.client.model.Document;
+import org.nuxeo.ecm.automation.client.model.Documents;
+import org.nuxeo.ecm.automation.client.model.FileBlob;
+import org.nuxeo.ecm.automation.core.operations.blob.AttachBlob;
+import org.nuxeo.ecm.automation.core.operations.blob.GetDocumentBlob;
+import org.nuxeo.ecm.automation.core.operations.document.CreateDocument;
+import org.nuxeo.ecm.automation.core.operations.document.FetchDocument;
+import org.nuxeo.ecm.automation.core.operations.services.bulk.AutomationBulkAction;
+import org.nuxeo.ecm.automation.core.operations.services.bulk.BulkRunAction;
+import org.nuxeo.ecm.automation.core.operations.services.query.DocumentPaginatedQuery;
+import org.nuxeo.ecm.automation.test.EmbeddedAutomationServerFeature;
+import org.nuxeo.ecm.core.bulk.CoreBulkFeature;
+import org.nuxeo.ecm.core.test.annotations.Granularity;
+import org.nuxeo.ecm.core.test.annotations.RepositoryConfig;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.ServletContainer;
+import org.nuxeo.runtime.test.runner.TransactionalFeature;
+
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletResponse;
+import java.io.File;
+import java.io.Serializable;
+import java.util.HashMap;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static junit.framework.TestCase.assertEquals;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+
+/**
+ * @since 10.3
+ */
+@RunWith(FeaturesRunner.class)
+@Features({ EmbeddedAutomationServerFeature.class, CoreBulkFeature.class })
+@Deploy("org.nuxeo.ecm.automation.test")
+@Deploy("org.nuxeo.ecm.automation.test:operation-contrib.xml")
+@Deploy("org.nuxeo.ecm.automation.test:test-bindings.xml")
+@ServletContainer(port = 18080)
+@RepositoryConfig(cleanup = Granularity.METHOD)
+public class AsyncOperationAdapterTest {
+
+    @Inject
+    Session session;
+
+    @Inject
+    public TransactionalFeature txFeature;
+
+    AsyncSession async;
+
+    @Before
+    public void setup() {
+        async = session.getAdapter(AsyncSession.class);
```

Comment:
We can add `AsyncSession` as an injectable field in `EmbeddedAutomationServerFeature#configure`, you would avoid the `@Before`.

Reply:
👌